services/system_undelete.proto

```diff
@@ -29,18 +46,43 @@ option java_multiple_files = true;
 import "basic_types.proto";
 
 /**
- * Undelete a file or smart contract that was deleted by SystemDelete; requires a Hedera
- * administrative multisignature. 
+ * Recover a file or contract bytecode deleted from the Hedera File
+ * System (HFS) by a `systemDelete` transaction.
+ *
+ * > Note
+ * >> A system delete/undelete for a `contractID` is not supported and
+ * >> SHALL return `INVALID_FILE_ID` or `MISSING_ENTITY_ID`.
+ *
+ * This transaction can _only_ recover a file removed with the `systemDelete`
+ * transaction. A file deleted via `fileDelete` SHALL be irrecoverable.<br/>
+ * This transaction MUST be signed by an Hedera administrative ("system")
+ * account.
+ *
+ * ### Block Stream Effects
+ * None
  */
 message SystemUndeleteTransactionBody {
     oneof id {
         /**
-         * The file ID to undelete, in the format used in transactions
+         * A file identifier.
+         * <p>
+         * The identified file MUST exist in the HFS.<br/>
+         * The identified file MUST be deleted.<br/>
+         * The identified file deletion MUST be a result of a
+         * `systemDelete` transaction.<br/>
```

Comment:
@povolev15 It does not look to me like we are checking that the file was previously deleted by a systemDelete transaction. Should we be checking this? Or should these documentation lines be removed?

Reply:
Given the content cannot be restored for a regular delete; I would expect that a system undelete from a regular delete simply won't work correctly (transaction might still "succeed", but perhaps _should_ not?).

Reply:
I think that the system_undelete handler code should be updated to check whether a file has been deleted by system_delete, and to error if it has not. I will file an issue for this.

Reply:
I checked with @Neeharika-Sompalli. She says that the current behavior (if file has not been deleted then do nothing, if file has been marked as deleted by either file_delete or system_delete then mark as not deleted) is correct.